src/commands/Tools/avatar.ts

```diff
@@ -13,20 +17,37 @@ export default class extends SkyraCommand {
 			extendedHelp: language => language.tget('COMMAND_AVATAR_EXTENDED'),
 			requiredPermissions: ['EMBED_LINKS'],
 			runIn: ['text'],
-			usage: '(user:username)'
+			usage: '[user:username]',
+			flagSupport: true
 		});
-
-		this.createCustomResolver('username', (arg, possible, msg) =>
-			arg ? this.client.arguments.get('username')!.run(arg, possible, msg) : msg.author);
 	}
 
-	public async run(message: KlasaMessage, [user]: [KlasaUser]) {
+	public async run(message: KlasaMessage, [user = message.author]: [KlasaUser]) {
 		if (!user.avatar) throw message.language.tget('COMMAND_AVATAR_NONE');
 
+		const extFlag = Reflect.get(message.flagArgs, 'ext');
+		const ext = extFlag ? this.resolveExt(extFlag) : 'png';
+
+		const sizeFlag = Reflect.get(message.flagArgs, 'size');
+		const size = sizeFlag ? this.resolveSize(sizeFlag) : 2048;
+
 		return message.sendEmbed(new MessageEmbed()
-			.setAuthor(user.tag, user.displayAvatarURL({ size: 128, format: 'png', dynamic: true }))
+			.setAuthor(user.tag, user.displayAvatarURL({ size: 128, format: ext, dynamic: true }))
 			.setColor(await DbSet.fetchColor(message))
-			.setImage(user.displayAvatarURL({ size: 2048, format: 'png', dynamic: true })));
+			.setImage(user.displayAvatarURL({ size, format: ext, dynamic: true })));
+	}
+
+	private resolveExt(ext: string): ImageExt {
+		const results = this.kExtensionRegex.exec(ext);
+		this.kExtensionRegex.lastIndex = 0;
+		if (results) return results[0].toLowerCase() as ImageExt;
```

Comment:
For a string like `png`, this is fine, but for one like `.png`, it isn't 
![image](https://user-images.githubusercontent.com/17960496/89797133-39eefc80-db33-11ea-94c8-05c11cfded34.png)

```suggestion
		if (results) return results[1].toLowerCase() as ImageExt;
```

Reply:
I've talked to Favna and I'm removing the extension flag entirely, so this will not matter.